src/lib/api.ts

```diff
@@ -0,0 +1,208 @@
+import axios, {
+    AxiosError,
+    AxiosInstance,
+    AxiosInterceptorManager,
+    AxiosPromise,
+    AxiosRequestConfig,
+    AxiosResponse,
+    ResponseType,
+} from 'axios';
+import pickBy from 'lodash/pickBy';
+import DownloadReachability from './DownloadReachability';
+import MetadataAPI from './metadataAPI';
+
+export type APIGetConfig = { type?: ResponseType } & AxiosRequestConfig;
+export type APIError = { response: AxiosResponse } & Error;
+export type APIPromise = Promise<AxiosPromise | SyntaxError>;
+
+/**
+ * Retrieves JSON from response.
+ *
+ * @private
+ * @param {Response} response - Response to parse
+ * @return {Promise|Response} Response if 204 or 202, otherwise promise that resolves with JSON
+ */
+const parseResponse = (response: AxiosResponse): AxiosResponse | AxiosResponse['data'] => {
+    if (response.status === 204 || response.status === 202) {
+        return response;
+    }
+
+    return response.data;
+};
+
+/**
+ * Filter empty values from the http request options object
+ *
+ * @private
+ * @param {Object} options - The request options
+ * @return {Object} The cleaned request options
+ */
+const filterOptions = (options: AxiosRequestConfig = {}): AxiosRequestConfig => {
+    return pickBy(options, (value: keyof AxiosRequestConfig) => value !== undefined && value !== null);
+};
+
+/**
+ * Helper function to convert an http error to the format Preview expects
+ *
+ * @private
+ * @param {Object} response - Axios error response
+ * @throws {Error} - Throws when an error response object exists
+ * @return {void}
+ */
+const handleError = ({ response }: AxiosError): void => {
+    if (response) {
+        const error = new Error(response.statusText) as APIError;
+        error.response = response; // Need to pass response through so we can see what kind of HTTP error this was
+        throw error;
+    }
+};
+
+/**
+ * Pass through transformer if the response type is text
+ * @param {Object} data
+ * @return {Object}
+ */
+const transformTextResponse = <D>(data: D): D => data;
+
+export default class Api {
```

Comment:
nit: normally we try to keep exports all at the bottom. Though I guess with types we have them wherever they are defined.